main.py

```diff
@@ -38,52 +40,60 @@ def __init__(self):
             exit('Check if the configuration is set right')
 
     def _run(self):
-        for submission in self.reddit.subreddit(self.config.SUBREDDIT).stream.submissions():
-            print(submission.title + " - http://reddit.com" + submission.permalink)
-
+        for submission in self._get_submissions():
             exceeds_threshold = False
+            parsed_submission = TitleParser.create_parsed_submission_from_submission(submission)
 
-            if self.database.known_submission(submission) is True:
-                continue
-
-            if submission.is_self is True:
-                self.database.save_submission(submission)
-                # We currently don't do anything further with self posts.  Move to the next post.
-                continue
-
-            try:
-                parsed_submission = TitleParser.create_parsed_submission_from_submission(submission)
-            except LookupError:
-                # Can't find artist from submission name, skipping
-                self.database.log_error_causing_submission(None, submission, traceback.format_exc(), True)
+            if ExcludedArtists.is_excluded(parsed_submission):
                 continue
 
             for service in self.services:
                 try:
+                    # Some services can run without a successful parsed submission and just need the url
+                    # If this is not the case, it will throw an exception and continue
                     result = service.get_service_result(parsed_submission)
 
-                    if result.exceeds_threshold is True:
-                        self._perform_exceeds_threshold_mod_action(submission, result)
-                        exceeds_threshold = True
+                    if result.exceeds_threshold is False:
```

Comment:
This might look clearer if the logic is back to 

`if result.exceeds_threshold:`

Instead of the continue we can put the actions within the if (including the break).